File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter05/LocalAsync/SimpleMath.cs
using System;
using System.Threading;

// In SimpleMath.cs
namespace MathLibrary
{
public class SimpleMath
{
   public int Add(int n1, int n2)
   {
      // Display the current thread id
      Console.WriteLine("SimpleMath.Add() executing on thread {0}", 
         Thread.CurrentThread.GetHashCode());

      // Waste some time
      Thread.Sleep(5000);

      return n1 + n2;
   }

      public int Subtract(int n1, int n2)
      {
         // Display the current thread id
         Console.WriteLine("SimpleMath.Subtract() executing on thread {0}", 
            Thread.CurrentThread.GetHashCode());

         // Waste some time
         Thread.Sleep(5000);

         return n1 - n2;
      }
   }
}

File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter05/AsyncCallback/SimpleMath.vb
Imports System.Threading

Public Class SimpleMath
   
   Public Function Add(n1 As Integer, n2 As Integer) As Integer
      
      'Display the current thread id
      Console.WriteLine("SimpleMath.Add() executing on thread {0}", _
         Thread.CurrentThread.GetHashCode())

      'Waste some time
      Thread.Sleep(5000)

      Return n1 + n2
   End Function
   
   Public Function Subtract(n1 As Integer, n2 As Integer) As Integer
      
      'Display the current thread id
      Console.WriteLine("SimpleMath.Subtract() executing on thread {0}", _
         Thread.CurrentThread.GetHashCode())

      'Waste some time
      Thread.Sleep(5000)
      
      Return n1 - n2
   End Function 
End Class